### PR TITLE
create tree path for artificial trees

### PR DESCRIPTION
**Pull request by txiang61**

### CODE REVIEW

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java [hunk 1]
```diff
@@ -1706,6 +1706,14 @@ public PhaseOneResult process(CompilationUnitTree root, UnderlyingAST underlying
          */
         public void handleArtificialTree(Tree tree) {}
 
+        /**
+         * Perform any actions required when CFG translation created a Tree that is not part of the
+         * original AST and added the TreePath.
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java:1711]: The `and added the TreePath` isn't clear at all. That is something that the translation already does? Or that is something that this method will do?
Making the distinction between the two versions clearer, and linking them with `@see` would be helpful.

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java [hunk 2]
```diff
@@ -2875,7 +2883,7 @@ public Node visitCompoundAssignment(CompoundAssignmentTree tree, Void p) {
                                         withoutAssignment(kind),
                                         tree.getVariable(),
                                         tree.getExpression());
-                        handleArtificialTree(operTree);
+                        handleArtificialTreeWithPath(operTree);
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java:2886]: What is the logic for changing these invocations and not others? There are still 32 calls of plain `handleArtificialTree` and I'm not sure I see the reason why some now have a Path and others don't.

File: framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java [hunk 3]
```diff
@@ -2999,6 +2999,10 @@ public final void setEnclosingElementForArtificialTree(Tree node, Element enclos
         artificialTreeToEnclosingElementMap.put(node, enclosing);
     }
 
+    public void creatTreePathForArtificialTree(Tree node, TreePath path) {
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java:3002]: Typo in method name `creat` -> `create`

Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java:3002]: Add some documentation.

**Comment by txiang61** [comment]
resolve in #133

**Pull request closed without merging**